beacon_node/eth2-libp2p/src/service.rs

```diff
@@ -363,3 +416,13 @@ fn load_private_key(config: &NetworkConfig, log: &slog::Logger) -> Keypair {
     }
     local_private_key
 }
+
+/// Generate authenticated XX Noise config from identity keys
+fn generate_noise_config(
+    identity_keypair: &Keypair,
+) -> noise::NoiseAuthenticated<noise::XX, noise::X25519, ()> {
+    let static_dh_keys = noise::Keypair::<noise::X25519>::new()
+        .into_authentic(identity_keypair)
+        .expect("signing can fail only once during starting a node");
```

Comment:
@AgeManning couldn't find a way to return this as a `Boxed` error type in `build_transport`. Felt it might be okay since it can only happen before the node starts during initialization of the libp2p service. Thoughts?

Reply:
I think this is alright to fail loudly if we cannot sign the keypair. 

We could propagate the error through to the box, but in the end it will fail if we cannot sign the key. So I think this is fine.